bit_manipulation/unique_in_array_xor.py

```diff
@@ -0,0 +1,13 @@
+def find_non_duplicate(nums: list) -> int:
+    '''
+    >>>find_non_duplicate([1, 2, 1, 2, 3, 4, 3])
```

Comment:
```suggestion
    >>> find_non_duplicate([1, 2, 1, 2, 3, 4, 3])
```
Fix test so it runs.

Reply:
Please add test.
>>> find_non_duplicate([1, 2, 3])